src/lib/setup/prompt.ts

```diff
@@ -1,15 +1,41 @@
 import fs from "fs";
 import inquirer from "inquirer";
+
 import {
   validateAccessToken,
   validateOrgName,
   validateProjectName,
   validateServicePrincipalId,
   validateServicePrincipalPassword,
-  validateServicePrincipalTenantId
+  validateServicePrincipalTenantId,
+  validateSubscriptionId
 } from "../validator";
 import { DEFAULT_PROJECT_NAME, IRequestContext, WORKSPACE } from "./constants";
 import { createWithAzCLI } from "./servicePrincipalService";
+import { getSubscriptions } from "./subscriptionService";
+
+export const promptForSubscriptionId = async (rc: IRequestContext) => {
+  const subscriptions = await getSubscriptions(rc);
+  if (subscriptions.length === 0) {
+    throw Error("no subscriptions found");
+  }
+  if (subscriptions.length === 1) {
+    rc.subscriptionId = subscriptions[0].id;
+  } else {
+    const questions = [
+      {
+        choices: subscriptions.map(s => s.name),
+        message: "Select one of the subscription\n",
+        name: "az_subscription",
+        type: "list"
+      }
+    ];
+    const ans = await inquirer.prompt(questions);
+    rc.subscriptionId = subscriptions.find(
+      s => s.name === ans.az_subscription
```

Comment:
I think adding a case insensitive would be good when checking for subscription names.

Reply:
this is not necessary. because the name is returned by SDK; and we are formulating the question based on these names and then match them with the id. Can you tell me why do not need to case insensitive check here?

Reply:
If user is selecting the name from a list of options, it is fine. I missed that part earlier. If user enters name, I would make case insensitive match in this scenario.